src/mlcubes/data_preparation/project/Dockerfile

```diff
@@ -0,0 +1,95 @@
+FROM ubuntu:18.04
```

Comment:
It could. I can move all the logic for the MLCube creation there and it would make things simpler (no need to clone the repository). It would no longer call the BratsPipeline directly as it currently does, since the entrypoint would be replaced with the MLCube entrypoint. If that's fine then I can make the necessary changes

Reply:
Sound good! I'll move the MLCube docker image creation to the root Dockerfile then

Reply:
Hey @sarthakpati, although this helps on having a more stable build procedure, it is now taking longer to build and I identified the source.

Docker builds images by layers, and caches every layer that hasn't changed in between build attempts. It would be ideal if compiling the binaries only happens when the source code of those binaries are modified, but at the moment any change on the repository will trigger the compilation step, which is not ideal.

That's because of this:
https://github.com/FeTS-AI/Front-End/blob/e116b8e304ac7a9f6bfd26fd78537498d837b5a3/Dockerfile#L12-L14

We're copying everything inside the root folder and then compiling the binaries. Ideally, we would only copy the files that are required for building the binaries, and after that we can copy everything else. Do you know if this is possible? I'm not familiar with what specific files are required during build time, so that's why I'm reaching back to you. Ideally we would have something like this

```Dockerfile
COPY required/files/for/build/time .

RUN mkdir bin && cd bin && cmake -DCMAKE_INSTALL_PREFIX="./install/appdir/usr" -DITK_DIR="/CaPTk/bin/ITK-build" -DDCMTK_DIR="/CaPTk/bin/DCMTK-build" -DBUILD_TESTING=OFF .. && make -j$(nproc) && make install/strip

COPY . .
```

Doing so would make image build time go down to seconds, instead of 15 minutes which is what is taking me at this moment